index.bs

```diff
@@ -12639,21 +12639,22 @@ internal method as follows.
     is used to determine if a given named property is exposed on an object.
     Some named properties are not exposed on an object depending on whether
     the [{{OverrideBuiltins}}] [=extended attribute=] was used.
-    The algorithm operates as follows, with property name |P| and object |O|:
+    The algorithm operates as follows, with property name |P|, object |O|, and
+    optionally object |propertyHolder|:
 
+    1.  If |propertyHolder| was not specified, set |propertyHolder| to |O|.
     1.  If |P| is not a [=supported property name=] of |O|, then return false.
-    1.  If |O| has an own property named |P|, then return false.
+    1.  If |propertyHolder| has an own property named |P|, then return false.
 
           Note: This will include cases in which |O| has unforgeable properties,
           because in practice those are always set up before objects have any supported property names,
           and once set up will make the corresponding named properties invisible.
 
     1.  If |O| implements an interface that has the [{{OverrideBuiltins}}]
         [=extended attribute=], then return true.
-    1.  Let |prototype| be |O|.\[[GetPrototypeOf]]().
+    1.  Let |prototype| be |propertyHolder|.\[[GetPrototypeOf]]().
     1.  While |prototype| is not null:
-        1.  If |prototype| is not a [=named properties object=],
-            and |prototype| has an own property named |P|, then return false.
```

Comment:
Are we sure that removing this clause is safe for the other 3 or so call sites of "named property visiblity algorithm"?

Reply:
It'll only affect the other call sites if someone takes an object that supports named properties and sticks the named properties object into its proto chain. I don't think we need to care particularly about that scenario.

Reply:
Well, it's observable, right? If so, what do implementations do?

Reply:
IIRC, it's only observable in Gecko, and Gecko matches the new spec text.